https://bugs.gentoo.org/946985
https://github.com/hughsie/libjcat/commit/130179f55a4eba1e885ca4ea1b4b2aa0a8d24605

From 130179f55a4eba1e885ca4ea1b4b2aa0a8d24605 Mon Sep 17 00:00:00 2001
From: Heiko Becker <mail@heiko-becker.de>
Date: Sat, 2 Nov 2024 15:06:31 +0100
Subject: [PATCH] Skip ed25519 part of a test with -Ded25519=false

Otherwise it fails with "jcat-self-test.c:1147:jcat_bt_common_func:
assertion failed (error == NULL): Jcat engine kind 'ed25519' not
supported (g-io-error-quark, 1)"
---
 libjcat/jcat-self-test.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libjcat/jcat-self-test.c b/libjcat/jcat-self-test.c
index 68e1273..cc92e12 100644
--- a/libjcat/jcat-self-test.c
+++ b/libjcat/jcat-self-test.c
@@ -1142,6 +1142,7 @@ jcat_bt_common_func(void)
 	g_assert_no_error(error);
 	g_assert_nonnull(btcheckpoint);
 
+#ifdef ENABLE_ED25519
 	/* get engine */
 	engine = jcat_context_get_engine(context, JCAT_BLOB_KIND_ED25519, &error);
 	g_assert_no_error(error);
@@ -1162,6 +1163,7 @@ jcat_bt_common_func(void)
 					   &error);
 	g_assert_no_error(error);
 	g_assert_nonnull(result);
+#endif
 }
 
 int

